### PR TITLE
added WithSliceDeepCopy config flag

### LINKED ISSUE #121: how to merge slice struct
```
package main

import (
	"fmt"
	"github.com/imdario/mergo"
)

type Order struct {
	A       string
	B       int64
	Details []Detail
}

type Detail struct {
	A string
	B string
}

func main() {
	src := Order{
		A: "one",
		B: 2,
		Details: []Detail{
			{
				B: "B",
			},
		},
	}
	dest := Order{
		A: "two",
		Details: []Detail{
			{
				A: "one",
			},
		},
	}
	mergo.Merge(&dest, src)
	fmt.Printf("%+v",dest)
	// Will print
	// {A:two B:2 Details:[{A:one B:}]}
	// how to get result
	// {A:two B:2 Details:[{A:one B:B}]}
}
```
Will print
```
{A:two B:2 Details:[{A:one B:}]}
```
I want to got 
```
{A:two B:2 Details:[{A:one B:B}]}
```

I want to merge two sorted Detail slice data

### PR DESCRIPTION
**Pull request by PeerXu**
This fixed #121

### CODE REVIEW

**Comment by dionysius** [comment]
Can we get that merged?

File: issue121_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	if err := Merge(&dst, src, WithSliceDeepCopy); err != nil {
+		t.Fatalf("Error during the merge: %v", err)
+	}
+
+	if dst["inter"].(map[string]interface{})["a"].(string) != "3" {
+		t.Error("inter.a should equal '3'")
+	}
+
+	if dst["inter"].(map[string]interface{})["c"].(string) != "4" {
```
Comment (zored) [inline issue121_test.go]: It would be great also to check `"b"` here

File: merge.go [hunk 2]
```diff
@@ -142,6 +144,24 @@ func deepMerge(dst, src reflect.Value, visited map[uintptr]*visit, depth int, co
 							return fmt.Errorf("cannot append two slices with different type (%s, %s)", srcSlice.Type(), dstSlice.Type())
 						}
 						dstSlice = reflect.AppendSlice(dstSlice, srcSlice)
+					} else if sliceDeepCopy {
+						i := 0
+						for ; i < srcSlice.Len() && i < dstSlice.Len(); i++ {
+							srcElement := srcSlice.Index(i)
```
Comment (zored) [inline merge.go]: What if slices lengths is different here and below?

**Comment by zored** [comment]
Great changes! Waiting it release!

File: merge.go [hunk 3]
```diff
@@ -261,6 +296,12 @@ func WithTypeCheck(config *Config) {
 	config.TypeCheck = true
 }
 
+// WithSliceDeepCopy will merge slice element one by one with Overwrite flag.
```
Comment (dionysius) [inline merge.go]: We should clearly note that it merges the same index. That way the user is informed that the order of the slices is important.

**Comment by imdario** [comment]
I'll check this ASAP.

File: issue121_test.go (hunk 1 above)
Comment (imdario) [reply issue121_test.go:30]: @PeerXu Can you check this, please? I would like to merge it.

**Comment by arnaudbriche** [comment]
Any news on this ? Will it be merged ?

**Comment by imdario** [comment]
Merged, but it doesn't overwrite. Overwrite can be added as an additional flag.

**Pull request merged**